File: ruby/balancing_parentheses.rb
def balancing_parentheses(string)
  arr = string.split('')
  count = 0
  arr.map{|i|
  if i == '('
    count-=1
  elsif i == ')'
    count+=1
  end
}
  count
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts balancing_parentheses('(()())')

  puts

  puts "Expecting: 2"
  puts balancing_parentheses('()))')

  puts

  puts "Expecting: 1"
  puts balancing_parentheses(')')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/balancing_parentheses.js
function balancingParentheses(string) {
  let arr = string.split('')
  let count = 0;
  arr.forEach(i=>{
    if (i==='('){
      count= count - 1
    } else if (i===')'){
      count+=1
    }
  })
  return count
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("Result: ", balancingParentheses('(()())'));

  console.log("");

  console.log("Expecting: 2");
  console.log("Result: ", balancingParentheses('()))'));

  console.log("");

  console.log("Expecting: 1");
  console.log("Result: ", balancingParentheses(')'));

  console.log("");

  console.log("Expecting: 0");
  console.log("Result: ", balancingParentheses(')('));

  console.log("");

  console.log("Expecting: 1");
  console.log("Result: ", balancingParentheses(')()'));
}

module.exports = balancingParentheses;

// Please add your pseudocode to this file
// And a written explanation of your solution
